### ISSUE TITLE
Find a reliable way to `dcast` empty `data.table`

### ISSUE THREAD

**Issue by pfistfl** [mlr-org/mlr3forecasting#23]
`expect_backend` and `train` etc. call `task$head(0L)`.
This currently breaks.

```r
tsk = mlr_tasks$get("AirPassengers")
tsk$head(0L)
```

**Comment by pfistfl** [comment]
There are various other reasons for why `expect_backend` does not pass:

```r
      #TODO: Hotfix #1:
      # Problem: We need a valid output for rows = NULL
      #TODO: Hotfix #2:
      # Problem: We need a valid output for cols == self$primary_key
      # TODO: Hotfix #3:
      # Problem: We need a valid output for cols == character()
      # Problem:
      # Rows 81-89 in helper_expectations.R do `-1` on the date which results in NA
      # somehow breaks the POSIXct type.
```

**Comment by pfistfl** [comment]
Additionally we check whether row_roles somewhere in `task_rm_data` are double which does not seem to work?

```r
Error: Assertion on 'elements of row_roles' failed: Must be of type 'double', not 'integer'
```

**Comment by pfistfl** [comment]
Closed via #25
